### PR TITLE
fix: misc fixes

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #[issue_no]

## Summary of Changes

These are a bunch of fixes I accumulated while working on some other stuff. None of them big enough for their own PR but I also didn't want to tie them into the cfms-15 lifetime.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Check that FMGC flight phases increment correctly on the MCDU. Check that attempting to clear the FROM or active waypoint on the F-PLN page gives a `NOT ALLOWED IN NAV` scratchpad message.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Utils/NXSystemMessages.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ const NXSystemMessages = {
     newCrzAlt:              new McduMessage("NEW CRZ ALT - HHHHH", false, true, "HHHHH"),
     noIntersectionFound:    new McduMessage("NO INTERSECTION FOUND", false, false),
     notAllowed:             new McduMessage("NOT ALLOWED", false, false),
+    notAllowedInNav:        new McduMessage("NOT ALLOWED IN NAV", false, false),
```
Comment (tracernz) [inline flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Utils/NXSystemMessages.js:50]: This message is used in the flight plan page but didn't exist.

File: src/fmgc/src/flightphase/index.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { FlightPhaseManager } from './FlightPhaseManager';
+
+const flightPhaseManager = new FlightPhaseManager();
+
+export {
+    FlightPhaseManager,
+}
+
+export function getFlightPhaseManager(): FlightPhaseManager {
```
Comment (tracernz) [inline src/fmgc/src/flightphase/index.ts:9]: This makes the flight phase manager a singleton, and available for use by typescript code (before it was only usable by the old js code).

File: src/fmgc/src/flightphase/FlightPhaseManager.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ function canInitiateDes(distanceToDestination: number): boolean {
 }
 
 export class FlightPhaseManager {
-    private activePhase = SimVar.GetSimVarValue('L:A32NX_INITIAL_FLIGHT_PHASE', 'number') || FmgcFlightPhase.Preflight;
+    private activePhase: FmgcFlightPhase = SimVar.GetSimVarValue('L:A32NX_INITIAL_FLIGHT_PHASE', 'number') || FmgcFlightPhase.Preflight;
```
Comment (tracernz) [inline src/fmgc/src/flightphase/FlightPhaseManager.ts:16]: Due to the simvar call the type couldn't be deduced before.

**Review by globin — COMMENTED** [review]
LGTM

**Review by davidwalschots — APPROVED** [review]

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #6758
Tier of Testing  : 1
Date             : 19/02/2022

Testing Process: 
Tested all flight phases and everything sequenced correctly.
Got NOT ALLOWED IN NAV when trying to clear FROM or active waypoint while in NAV.

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Pull request merged**